File: docs/examples/python/output/parts/UserManagementService_find_by_email_domain.py
# code2map fragment (non-buildable)
# id: CD12
# original: docs/examples/python/user_management_service.py
# lines: 177-192
# symbol: UserManagementService#find_by_email_domain
# notes: references dataclasses.dataclass, re, typing.Optional; calls result.append, self._users.values, user.email.endswith
    def find_by_email_domain(self, domain: str) -> list[User]:
        """
        メールドメインでユーザーを検索する

        Args:
            domain: メールドメイン（例: "example.com"）

        Returns:
            条件に一致するユーザーのリスト
        """
        result = []
        domain_suffix = f"@{domain}"
        for user in self._users.values():
            if user.email.endswith(domain_suffix):
                result.append(user)
        return result

File: code2map/docs/examples/java/output/parts/UserManagementService_findByEmailDomain.java
// code2map fragment (non-buildable)
// original: docs/examples/java/UserManagementService.java
// lines: 146-155
// symbol: UserManagementService#findByEmailDomain
// notes: references java.util.ArrayList, java.util.HashMap, java.util.List, java.util.Map, java.util.Optional, java.util.regex.Pattern; calls endsWith, result.add, user.getEmail, users.values
    public List<User> findByEmailDomain(String domain) {
        List<User> result = new ArrayList<>();
        String domainSuffix = "@" + domain;
        for (User user : users.values()) {
            if (user.getEmail().endsWith(domainSuffix)) {
                result.add(user);
            }
        }
        return result;
    }
